app/services/school_data_importer.rb

```diff
@@ -0,0 +1,70 @@
+require "open-uri"
+require "csv"
+
+class SchoolDataImporter
+  def run
+    download_and_save_csv
+
+    CSV.foreach(csv_file_name, headers: true, encoding: "windows-1251:utf-8") do |row|
+      local_authority = LocalAuthority.find_or_create_by(
+        code: row["LA (code)"],
+        name: row["LA (name)"]
+      )
+
+      school = row_to_school(row)
+
+      school.local_authority = local_authority
+
+      unless school.valid?
+        Rails.logger.warn("Failed to import school: #{school.urn} - #{school.errors.full_messages}") && next
+      end
+      school.save
```

Comment:
I would be tempted to call `save!` here and not warn on validation failures. If there is data that doesn't validate in our code we definitely want a noisy warning about that. And if it's the source data that is dodgy, we we also probably want to know about that and then make a call on whether we want to skip over schools that don't validate.